chain/rust/src/fees.rs

```diff
@@ -50,6 +62,67 @@ pub fn min_script_fee(
     }
 }
 
+
+
+// pub fn total_ref_scripts_size(ref_inputs: &[TransactionUnspentOutput]) -> Result<u64, ArithmeticError> {
+//     ref_inputs
+//         .iter()
+//         .try_fold(0, |acc, utxo|
+//             utxo
+//                 .output
+//                 .script_ref()
+//                 .map(|script| script.to_cbor_bytes().len() as u64)
+//                 .unwrap_or(0u64)
+//                 .checked_add(acc)
+//                 .ok_or(ArithmeticError::IntegerOverflow))
+// }
+
+/**
+ * Calculates the cost of all ref scripts
+ * * `total_ref_script_size` - Total size (original, not hashes) of all ref scripts. Duplicate scripts are counted as many times as they occur
+ */
+pub fn min_ref_script_fee(
+    tx: &Transaction,
+    linear_fee: &LinearFee,
+    total_ref_script_size: u64,
+) -> Result<Coin, ArithmeticError> {
+    // based on:
+    // https://github.com/IntersectMBO/cardano-ledger/blob/7e65f0365eef647b9415e3fe9b3c35561761a3d5/eras/conway/impl/src/Cardano/Ledger/Conway/Tx.hs#L84
+    // https://github.com/IntersectMBO/cardano-ledger/blob/a34f878c56763d138d2203d8ba84b3af64d94fce/eras/conway/impl/src/Cardano/Ledger/Conway/UTxO.hs#L152
+    use num::{rational::Ratio, CheckedAdd, CheckedMul};
+
```

Comment:
we might need `Ratio<BigNum>` during calculations then conversion afterwards

Reply:
Done, but do we care about other usage of the fraction crate? I changed this here since it just panics even though the final operation could fall into a `u64` fine if the intermediate denom/numerators don't. The other place we use the fraction crate is in genesis parsing.